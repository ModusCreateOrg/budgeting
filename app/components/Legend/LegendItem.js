// @flow
import * as React from 'react';
import formatAmount from 'utils/formatAmount';
import styles from './styles.scss';

type LegendItemProps = {
  color: string,
  value: number,
  label: string,
};

const LegendItem = ({ color, label, value }: LegendItemProps) => (
  <li style={{ color }}>
    <span>{label}</span>
    <span className={styles.value}> {formatAmount(value).text} </span>
  </li>
);

export default LegendItem;
