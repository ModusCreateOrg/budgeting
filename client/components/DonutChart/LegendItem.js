import React, { PropTypes } from 'react';

import formatAmount from 'utils/formatAmount';
import styles from './styles.scss';

const LegendItem = ({ color, label, value }) => (
  <li style={{ color }}>
    <span>{label}</span>
    <span className={styles.value}> {formatAmount(value).text} </span>
  </li>
);

LegendItem.propTypes = {
  color: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};

export default LegendItem;
