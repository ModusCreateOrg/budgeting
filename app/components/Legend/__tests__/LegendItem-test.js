import React from 'react';
import renderer from 'react-test-renderer';
import LegendItem from '../LegendItem';

it('renders correctly', () => {
  const tree = renderer.create(<LegendItem color="test" value={3} label="test" />).toJSON();
  expect(tree).toMatchSnapshot();

  window.requestAnimationFrame(() => 'wow');
});
