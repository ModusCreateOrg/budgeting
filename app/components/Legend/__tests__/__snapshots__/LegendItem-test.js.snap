// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  style={
    Object {
      "color": "test",
    }
  }
>
  <span>
    test
  </span>
  <span
    className="value"
  >
     
    $3.00
     
  </span>
</li>
`;
